python/ml4ir/base/model/scoring/scoring_model.py

```diff
@@ -381,6 +388,9 @@ def test_step(self, data):
             y = self.interaction_model.label_transform_op(y, training=False)
 
         y_pred = self(X, training=False)[self.output_name]
+        for _ in range(self.monte_carlo_inference_trials):  # MC trials during test.
+            y_pred += self(X, training=True)[self.output_name]
+        y_pred /= (self.monte_carlo_inference_trials+1)
```

Comment:
@mohazahran we should move this change out to a different Layer or wrapper. I assume this is a first pass direct port of your spike code?